File: src/main/kotlin/com/planner/weekly/api/v1/ReviewController.kt
package com.planner.weekly.api.v1

import com.planner.weekly.domain.review.WeeklyReview
import com.planner.weekly.domain.review.WeeklyReviewService
import com.planner.weekly.dto.response.ApiResponse
import com.planner.weekly.security.UserPrincipal
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.security.SecurityRequirement
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/plans/{planId}/review")
@Tag(name = "Weekly Review", description = "Weekly review APIs")
@SecurityRequirement(name = "bearerAuth")
class ReviewController(
    private val weeklyReviewService: WeeklyReviewService
) {

    @GetMapping
    @Operation(summary = "Generate weekly review for a plan")
    fun getReview(
        @AuthenticationPrincipal user: UserPrincipal,
        @PathVariable planId: String
    ): ApiResponse<WeeklyReview> {
        val review = weeklyReviewService.generateReview(planId, user.id)
        return ApiResponse.ok(review)
    }
}

File: src/review/review.controller.ts
import { Controller, Get, Param } from '@nestjs/common';
import { ApiTags, ApiOperation, ApiBearerAuth } from '@nestjs/swagger';
import { ReviewService } from './review.service';
import { WeeklyReviewResponseDto } from './dto/review.dto';
import { CurrentUser } from '../common/decorators/current-user.decorator';
import { JwtPayload } from '../common/interfaces/request-with-user.interface';
import { ApiResponse } from '../common/dto/api-response.dto';

@ApiTags('Weekly Review')
@ApiBearerAuth()
@Controller('plans/:planId/review')
export class ReviewController {
  constructor(private reviewService: ReviewService) {}

  @Get()
  @ApiOperation({ summary: '주간 회고 생성' })
  async getReview(
    @CurrentUser() user: JwtPayload,
    @Param('planId') planId: string,
  ): Promise<ApiResponse<WeeklyReviewResponseDto>> {
    const review = await this.reviewService.generateReview(planId, user.sub);
    return ApiResponse.ok(review);
  }
}
